api/src/main/java/org/apache/iceberg/actions/MigrateTable.java

```diff
@@ -44,6 +44,12 @@
    */
   MigrateTable tableProperty(String name, String value);
 
+  /**
+   * @param numReaders the number of concurrent file read operations to use per partition
+   * @return this for method chaining
+   **/
+  MigrateTable withParallelReads(int numReaders);
```

Comment:
Same as with SnapshotTable, default to unsupported so we can review just 3.2. Then merge the other modules in a followup.